### PR TITLE
journal: report disk limit timeout errors

### PR DESCRIPTION
**Pull request by strib**
This vendors keybase/client#6981 (in review).

This PR expands the existing `ErrDiskLimitTimeout` error to include usage information, and reports all that info back up to the service (throttling the errors to at most one per minute).

Issue: KBFS-2158

### CODE REVIEW

File: libkbfs/reporter_kbpki.go [hunk 1]
```diff
@@ -99,7 +102,7 @@ func (r *ReporterKBPKI) ReportErr(ctx context.Context,
 	params := make(map[string]string)
 	filename := ""
 	var code keybase1.FSErrorType = -1
-	switch e := err.(type) {
+	switch e := errors.Cause(err).(type) {
```
Comment (strib) [inline libkbfs/reporter_kbpki.go]: Oops!  Hopefully none of these errors were using the `errors` package...

File: libkbfs/tlf_journal.go [hunk 2]
```diff
@@ -1693,9 +1698,12 @@ func (j *tlfJournal) putBlockData(
 	case nil:
 		// Continue.
 	case context.DeadlineExceeded:
+		usageBytes, limitBytes, usageFiles, limitFiles :=
+			j.diskLimiter.getDiskLimitInfo()
```
Comment (strib) [inline libkbfs/tlf_journal.go]: There's a tiny race here, and it's possible that this call won't show usage being over the limit if a flush completed between the acquire and now, but I don't think it's worth changing the signature of `beforeBlockPut` for.  But what do you think?

File: libkbfs/block_util.go [hunk 3]
```diff
@@ -38,25 +39,42 @@ func putBlockToServer(ctx context.Context, bserv BlockServer, tlfID tlf.ID,
...
+		}
+	case ErrDiskLimitTimeout:
+		// Report this here in case the put is happening in a
+		// background goroutine (via `SyncAll` perhaps) and wouldn't
+		// otherwise be reported.  Mark the error as unreportable to
+		// avoid the upper FS layer reporting it twice, if this block
+		// put is the result of a foreground fsync.
+		reporter.ReportErr(ctx, tlfName, tlfID.IsPublic(), WriteMode, err)
+		typedErr.reportable = false
+		// Unfortunately this makes a different stack, but I don't
```
Comment (akalin-keybase) [inline libkbfs/block_util.go]: I think there's an easyish way:

1) Change `ErrDiskLimitTimeout.Error` to `(*ErrDiskLimitTimeout).Error` (i.e., make it take a pointer receiver);
2) then change `case ErrDiskLimitTimeout` to `case *ErrDiskLimitTimeout` everywhere;
3) then you'll be able to modify `typedErr` and return the original error (I think).
4) Probably worth a comment in `(*ErrDiskLimitTimeout).Error` explaining the reasoning.

File: libkbfs/block_util.go [hunk 4]
```diff
@@ -38,25 +39,42 @@ func putBlockToServer(ctx context.Context, bserv BlockServer, tlfID tlf.ID,
...
+		// think there's a way to preserve the old one and still
+		// change `reportable` field.
+		err = errors.WithStack(typedErr)
```
Comment (akalin-keybase) [inline libkbfs/block_util.go]: in this case you set `err`, but in the previous one you return the (nil) `err` directly. Perhaps make them consistent?

File: libkbfs/tlf_journal.go (hunk 2 above)
Comment (akalin-keybase) [inline libkbfs/tlf_journal.go]: Yeah, probably not worth it. Maybe add a comment, though?

**Pull request merged**